auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -117,6 +122,45 @@ internal suspend fun <T> OidcConfiguration.internalPerformRequest(
     }
 }
 
+private suspend fun OidcConfiguration.executeRequest(
+    request: Request
+): Response {
+    currentCoroutineContext().ensureActive()
+    val rateLimitErrorCode = 429
+
+    var response = okHttpClient.newCall(request).await()
+
+    if (response.code == rateLimitErrorCode) {
+        val event = RateLimitExceededEvent(request, response)
+        eventCoordinator.sendEvent(event)
+
+        for (i in 1..event.maxRetries) {
+            val timeToResetSeconds: Long
+            val requestId: String?
+            response.use {
+                val responseTimeSeconds = response.headers.getDate("Date")?.time?.milliseconds?.inWholeSeconds
+                val retryTimeSeconds = response.header("X-Rate-Limit-Reset")?.toLong()
```

Comment:
Can you add a test where this is malformed? For example the request id is `abcde`